core/src/main/scala/org/broadinstitute/dsde/rawls/util/WorkspaceSupport.scala

```diff
@@ -143,6 +143,21 @@ trait WorkspaceSupport {
     }
   }
 
+  def withWorkspaceBucketRegionCheck[T](bucketRegion: Option[String])(op: => Future[T]): Future[T] = {
+    bucketRegion match {
+      case Some(region) =>
+        // if the user specifies a region for the workspace bucket, it must be in the proper format for a single region
+        val singleRegionPattern = "[A-Za-z]+-[A-Za-z]+[0-9]+"
+        if (region.matches(singleRegionPattern)) op
+        else {
+          val err = ErrorReport(statusCode = StatusCodes.BadRequest, message = s"Workspace bucket location must be a single " +
+            s"(not multi-) region of format: $singleRegionPattern.")
```

Comment:
TOL still a little scary IMO. But I guess if we're only expecting this to be used by API power-users then that's ok (because most users would use a neat little UI element)

Reply:
Yes. Once we open this to all the users we can refine this and as you said with a neat UI element